openedx/adg/admin_dashboard/applications/tests/test_helpers.py

```diff
@@ -0,0 +1,136 @@
+"""
+Tests for all the helper functions in `admin_dashboard.applications` app.
+"""
+import json
+from datetime import datetime, timedelta
+
+import pytest
+
+from common.djangoapps.student.tests.factories import GroupFactory
+from openedx.adg.admin_dashboard.applications.constants import WEEKLY_SUBMISSIONS_GRAPH_DATE_FORMAT
+from openedx.adg.admin_dashboard.applications.helpers import (
+    get_new_applications_count_and_percentage_change,
+    get_submitted_applications,
+    get_weekly_application_submissions
+)
+from openedx.adg.admin_dashboard.applications.tests.constants import (
+    BUSINESS_LINE_TEST_NAME_1,
+    BUSINESS_LINE_TEST_NAME_2
+)
+from openedx.adg.lms.applications.constants import ADG_ADMIN_GROUP_NAME
+from openedx.adg.lms.applications.models import UserApplication
+from openedx.adg.lms.applications.tests.factories import (
+    ApplicationHubFactory,
+    BusinessLineFactory,
+    UserApplicationFactory,
+    UserFactory
+)
+from openedx.core.djangolib.testing.utils import skip_unless_lms
+
+
+def create_applications(number_of_applications, date):
+    """
+    Creates applications equal to the `number_of_applications` with `submission_date` set to `date` sent as argument.
+
+    Arguments:
+        number_of_applications (int): Depicts the number of applications to be created
+        date (datetime): Represents the date for application submission
+    """
+    for _ in range(number_of_applications):
+        user = UserFactory()
+        UserApplicationFactory(user=user)
+        ApplicationHubFactory(user=user, submission_date=date)
+
+
+@skip_unless_lms
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    'is_superuser, group_name, expected_count',
+    [
+        (True, None, 2),
+        (False, ADG_ADMIN_GROUP_NAME, 2),
+        (False, BUSINESS_LINE_TEST_NAME_1, 1)
+    ],
+    ids=['super_user', 'adg_admin', 'business_line_admin']
+)
+def test_get_submitted_applications(is_superuser, group_name, expected_count):
+    """
+    Test `get_submitted_applications` function for different types of admins.
+    """
+    user_1 = UserFactory()
+    user_2 = UserFactory()
+
+    business_line_1 = BusinessLineFactory(title=BUSINESS_LINE_TEST_NAME_1)
+    business_line_2 = BusinessLineFactory(title=BUSINESS_LINE_TEST_NAME_2)
+
```

Comment:
No need to create this explicitly, UserApplicationFactory will do this itself.

Reply:
I have a test case to verify where an admin of certain business line should only be shown application(s) from that business line hence I specified the title.